### DISCUSSION TITLE
Is Textual suitable for arrow key navigation?

### DISCUSSION THREAD

**Discussion by braindevices** [Textualize/textual#4863] [Q&A] [answered]
our environment normally do not provide mouse and need pure keyboard navigation through the TUI.
we have been using urwid, there the navigation by key is enabled by default.
But all the textual example seems only usable when there is a mouse, unless there is natural key binding like calculator.
I wonder is this library suitable for this kind of application at all.

**Comment by TomJGooding** [comment, marked as answer]
What makes you think that all the examples only seem usable with a mouse?

You can interact with all Textual widgets using the keyboard and use <kbd>Tab</kbd> to move focus to the next widget. Pressing <kbd>Shift</kbd>+<kbd>Tab</kbd> moves the focus in the opposite direction.

**Reply by braindevices** [reply]
thanks! we did not noticed this because we have tab and shift+tab binding in our DE

**Comment by willmcgugan** [comment]
@braindevices As Tom pointed out you can use `tab` and `shift`+`tab` to navigate controls in all Textual apps. You can also bind keys to do whatever  you like.
